package/sbp_protocol/src/sbp_router_smoothpose.yml

```diff
@@ -0,0 +1,169 @@
+      - dst_port: SBP_PORT_FILEIO_EXTERNAL
+        filters:
+          - { action: ACCEPT, prefix: [0x55, 0xA8, 0x00] } # File read
+          - { action: ACCEPT, prefix: [0x55, 0xA9, 0x00] } # File read dir
+          - { action: ACCEPT, prefix: [0x55, 0xAC, 0x00] } # File reamove
+          - { action: ACCEPT, prefix: [0x55, 0xAD, 0x00] } # File write
+          - { action: REJECT }
+      - dst_port: SBP_PORT_INTERNAL
+        filters:
+          - { action: ACCEPT }
+      - dst_port: SBP_PORT_SETTINGS_CLIENT
+        filters:
+          - { action: ACCEPT, prefix: [0x55, 0xA0, 0x00] } # Settings write
+          - { action: REJECT }
+
+  - name: SBP_PORT_FILEIO_FIRMWARE
+    pub_addr: "@tcp://127.0.0.1:43040"
+    sub_addr: "@tcp://127.0.0.1:43041"
+    forwarding_rules:
+      - dst_port: SBP_PORT_FIRMWARE
+        filters:
+          - { action: ACCEPT }
+
+  - name: SBP_PORT_FILEIO_EXTERNAL
+    pub_addr: "@tcp://127.0.0.1:43050"
+    sub_addr: "@tcp://127.0.0.1:43051"
+    forwarding_rules:
+      - dst_port: SBP_PORT_EXTERNAL
+        filters:
+          - { action: ACCEPT }
+
+  - name: SBP_PORT_INTERNAL
+    pub_addr: "@tcp://127.0.0.1:43060"
+    sub_addr: "@tcp://127.0.0.1:43061"
+    forwarding_rules:
+      - dst_port: SBP_PORT_EXTERNAL
+        filters:
+          - { action: ACCEPT }
+
+  - name: SBP_PORT_SETTINGS_CLIENT
+    pub_addr: "@tcp://127.0.0.1:43070"
+    sub_addr: "@tcp://127.0.0.1:43071"
+    forwarding_rules:
+      - dst_port: SBP_PORT_EXTERNAL
+        filters:
+          - { action: ACCEPT, prefix: [0x55, 0xAF, 0x00] } # Settings write response
+          - { action: REJECT }
+      - dst_port: SBP_PORT_SETTINGS_DAEMON
+        filters:
+          - { action: ACCEPT }
+      - dst_port: SBP_PORT_SETTINGS_CLIENT
+        filters:
+          - { action: ACCEPT, prefix: [0x55, 0xAF, 0x00] } # Settings write response
+          - { action: REJECT }
+
+  - name: SBP_PORT_SKYLARK
+    pub_addr: "@tcp://127.0.0.1:43080"
+    sub_addr: "@tcp://127.0.0.1:43081"
+    forwarding_rules:
+      - dst_port: SBP_PORT_FIRMWARE
+        filters:
+          - { action: ACCEPT, prefix: [0x55, 0x44, 0x00] } # SBP_MSG_BASE_POS_LLH
+          - { action: ACCEPT, prefix: [0x55, 0x48, 0x00] } # SBP_MSG_BASE_POS_ECEF
+          - { action: ACCEPT, prefix: [0x55, 0x4A, 0x00] } # SBP_MSG_OBS
+          - { action: ACCEPT, prefix: [0x55, 0x86, 0x00] } # SBP_MSG_EPHEMERIS_GPS
+          - { action: ACCEPT, prefix: [0x55, 0x88, 0x00] } # SBP_MSG_EPHEMERIS_GLO
+          - { action: ACCEPT, prefix: [0x55, 0x84, 0x00] } # SBP_MSG_EPHEMERIS_SBAS
+          - { action: ACCEPT, prefix: [0x55, 0x75, 0x00] } # SBP_MSG_GLO_BIASES
+          - { action: REJECT }
+
+  - name: SBP_PORT_NAV_DAEMON
+    pub_addr: "@tcp://127.0.0.1:43090"
+    sub_addr: "@tcp://127.0.0.1:43091"
+    forwarding_rules:
+      - dst_port: SBP_PORT_EXTERNAL
+        filters:
+          - { action: REJECT, prefix: [0x55, 0xAE, 0x00] }   # Settings register
```

Comment:
Why are we rejecting these?  If SmoothPose is going to read settings it'll need to register a different set of pub/sub sockets (which is feasible). If we want this port to just be for modified solution messages, then we should explicitly "ACCEPT" those messages and reject all others.

Reply:
Good catch.

I agree that we should just accept the following messages and reject anything else to external.  I'm putting the list here for my own record keeping and will update it as I brainstorm.

MSG_LOG 
MSG_INS_STATUS
MSG_POS_LLH
MOS_POS_LLH_COV
MSG_POS_ECEF
MSG_POS_ECEF_COV
MSG_VEL_NED
MSG_VEL_NED_COV
MSG_VEL_ECEF
MSG_VEL_ECEF_COV
MSG_VEL_BODY
MSG_ANGULAR_RATES
MSG_ORIENT_EULER
MSG_ORIENT_QUAT

MSG_SETTINGS_READ_RESP and MSG_SETTINGS_WRITE_RESP will be emitted by settings daemon for any settings owned by smoothpose so will be filtered out.

Reply:
👍 agree with this approach. only rerouted and nav only output messages should be whitelisted here. write responses are sent by the settings owner, but if they are using the relevant settings apis in libpiksi, this will all occur via the settings client endpoint